public/app/features/datasources/state/navModel.ts

```diff
@@ -53,14 +54,19 @@ export function buildNavModel(dataSource: DataSourceSettings, plugin: GenericDat
 
   const isLoadingNav = dataSource.type === loadingDSType;
 
-  const dsPermissions = {
+  const dsPermissions: NavModelItem = {
     active: false,
     icon: 'lock',
     id: `datasource-permissions-${dataSource.uid}`,
     text: 'Permissions',
     url: `datasources/edit/${dataSource.uid}/permissions`,
   };
 
+  if (highlightTrial()) {
+    dsPermissions.tabSuffix = () =>
+      ProBadge({ experimentId: 'feature-highlights-data-source-permissions-badge', eventVariant: 'trial' });
```

Comment:
The `experimentId` could go into a constant to avoid making mistakes and having two different IDs for trial and free users.